test/e2e/storage/persistent_volumes-disruptive.go

```diff
@@ -272,13 +272,16 @@ func kubeletCommand(kOp kubeletOpt, c clientset.Interface, pod *v1.Pod) {
 		framework.LogSSHResult(sshResult)
 	}
 	Expect(sshResult.Code).To(BeZero(), "Failed to [%s] kubelet:\n%#v", string(kOp), sshResult)
-	// On restart, waiting for node NotReady prevents a race condition where the node takes a few moments to leave the
-	// Ready state which in turn short circuits WaitForNodeToBeReady()
-	if kOp == kStop || kOp == kRestart {
+	if kOp == kStop {
 		if ok := framework.WaitForNodeToBeNotReady(c, pod.Spec.NodeName, NodeStateTimeout); !ok {
 			framework.Failf("Node %s failed to enter NotReady state", pod.Spec.NodeName)
 		}
 	}
+	if kOp == kRestart {
+		// On restart, Waiting for a minute before checking Node's Status to be Ready
```

Comment:
Should we also verify that kubelet's PID changed?

Reply:
@msau42 added code to verify if kublet PID is getting changed upon kubelet restart.